oneview_redfish_toolkit/util.py

```diff
@@ -408,3 +410,37 @@ def generate_certificate(dir_name, file_name, key_length, key_type="rsa"):
     with open(os.path.join(dir_name, file_name + ".key"), "wt") as f:
         f.write(OpenSSL.crypto.dump_privatekey(
             OpenSSL.crypto.FILETYPE_PEM, private_key).decode("UTF-8"))
+
+
+def dispatch_event(event):
+    """Execute the POST request to the targets of an event subscription
+
+        Args:
+            event: The Event schema describing the JSON payload
+            which will be send to the event subscribers.
+
+        Returns:
+            Nothing
+    """
+    # Event resource contains only ONE EventRecord
+    events = event.redfish['Events']
+    event_record = events[0]
+
+    subscriptions = \
+        globals()['subscriptions_by_type'][event_record['EventType']].values()
+
+    try:
+        for subscription in subscriptions:
+            url = urlparse(subscription.redfish['Destination'])
+            connection = HTTPConnection(url.hostname, port=url.port)
+            json_str = event.serialize()
+
+            connection.request(
+                'POST',
+                url.path,
+                json_str,
+                {'Content-Type': 'application/json'})
+    except ConnectionError as e:
```

Comment:
We should add retries as specified in the config file.
[event_service]
DeliveryRetryAttempts = 3
DeliveryRetryIntervalSeconds = 30 

Thanks

Reply:
Done! Please, check the file `event_dispatcher.py`